### PR TITLE
Ensures there is no slash in url before query params

### PR DESCRIPTION
**Pull request by ygrishajev**
Fixes reduxjs/redux-toolkit#2379

### CODE REVIEW

File: packages/toolkit/src/query/tests/utils.test.ts [hunk 1]
```diff
@@ -99,6 +101,10 @@ describe('joinUrls', () => {
     expect(joinUrls('https://example.com/api/', '/banana/')).toBe(
       'https://example.com/api/banana/'
     )
+
+    expect(joinUrls('https://example.com/api/', '?foo=bar')).toBe(
+      'https://example.com/api?foo=bar'
+    )
```
Comment (phryneas) [inline packages/toolkit/src/query/tests/utils.test.ts:107]: I'm too late to the party :(

I feel like this scenario is a bug. There are servers that expect the slash and will lead you to a completely different page without it - completely independently of a search param being present.

It's fine to *not add* the slash - but actively removing ti is dangerous.

**Comment by phryneas** [comment]
Could you please open another PR that keeps the slash if it is user-specified and only *not adds* it? Sorry for the hassle :sweat_smile:

**Comment by ygrishajev** [comment]
@phryneas I see, so you mean some user specified parameter on `createApi` function?

**Comment by phryneas** [comment]
On `fetchBaseQuery` - if `baseUrl` ends on a `/`, that should always stay in - likewise if `query` returns `{ url: 'foo/', params }`, the slash should stay as well, since the user explicitly specified it. It should just not appear out of the blue.

**Comment by markerikson** [comment]
I'm gonna revert this for now - can you file a new PR?

**Pull request merged**